commit a9c40a16bd5f5a2a7f5efa1c5f6d50bfba9344f4 (refs/changes/88/3188/3)
Author: Jared Morrow <jm@joyent.com>
Date:   2018-01-11T14:14:47-07:00 (1 year, 9 months ago)
    
    MANTA-3553 Muppet should recover when Zookeeper returns after being down
    Reviewed by: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
    Approved by: Kelly McLaughlin <kelly.mclaughlin@joyent.com>

diff --git a/muppet.js b/muppet.js
index 6ab8d19..f7f8290 100644
--- a/muppet.js
+++ b/muppet.js
@@ -284,10 +284,11 @@ function zookeeper(cfg) {
     /*
      * zkstream attempts to retry the connection to zookeeper. On retry
      * exhaustion it will emit a `failed` event.
+     * For this we only log an error, because when ZK comes back up, zkstream
+     * will automatically reconnect and re-arm the watchers (if they existed)
      */
     zk_client.on('failed', function onFailed(err) {
-        cfg.log.fatal(err, 'ZooKeeper: received failed event');
-        process.exit(1);
+        cfg.log.error(err, 'ZooKeeper: received failed event');
     });
 }
 
